frontend/src/employee-frontend/components/messages/MessageEditor.tsx

```diff
@@ -1,105 +1,232 @@
-import React, { useEffect, useState } from 'react'
-import styled from 'styled-components'
-import { faTimes, faTrash } from 'lib-icons'
-import colors from 'lib-customizations/common'
-import InputField from 'lib-components/atoms/form/InputField'
+import { Result } from 'lib-common/api'
+import { UpdateStateFn } from 'lib-common/form-state'
+import { UUID } from 'lib-common/types'
+import { useDebounce } from 'lib-common/utils/useDebounce'
+import { useRestApi } from 'lib-common/utils/useRestApi'
+import Button from 'lib-components/atoms/buttons/Button'
 import IconButton from 'lib-components/atoms/buttons/IconButton'
 import InlineButton from 'lib-components/atoms/buttons/InlineButton'
-import Button from 'lib-components/atoms/buttons/Button'
-import { defaultMargins, Gap } from 'lib-components/white-space'
+import InputField from 'lib-components/atoms/form/InputField'
+import MultiSelect from 'lib-components/atoms/form/MultiSelect'
+import Radio from 'lib-components/atoms/form/Radio'
+import { FixedSpaceRow } from 'lib-components/layout/flex-helpers'
 import { Label } from 'lib-components/typography'
+import { defaultMargins, Gap } from 'lib-components/white-space'
+import colors from 'lib-customizations/common'
+import { faTimes, faTrash } from 'lib-icons'
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import styled from 'styled-components'
+import Select from '../../components/common/Select'
 import { useTranslation } from '../../state/i18n'
-import { MessageBody } from './types'
+import * as api from './api'
 import {
+  deselectAll,
+  getReceiverOptions,
   getSelected,
   getSelectedBottomElements,
-  SelectorChange,
   SelectorNode,
   updateSelector
-} from 'employee-frontend/components/messages/SelectorNode'
-import MultiSelect from 'lib-components/atoms/form/MultiSelect'
-import Select from 'employee-frontend/components/common/Select'
-import Radio from 'lib-components/atoms/form/Radio'
-import { FixedSpaceRow } from 'lib-components/layout/flex-helpers'
-import { UUID } from 'lib-common/types'
+} from './SelectorNode'
+import { DraftContent, MessageBody, UpsertableDraftContent } from './types'
 
-const emptyMessageBody: MessageBody = {
-  title: '',
-  content: '',
-  type: 'MESSAGE',
-  recipientAccountIds: []
+type Message = UpsertableDraftContent & {
+  senderId: UUID
+  recipientAccountIds: UUID[]
 }
 
+const getInitialMessage = (
+  draft: DraftContent | undefined,
+  senderId: UUID
+): Message => ({
+  senderId,
+  content: draft?.content ?? '',
+  recipientIds: draft?.recipientIds ?? [],
+  recipientNames: draft?.recipientNames ?? [],
+  recipientAccountIds: [],
+  title: draft?.title ?? '',
+  type: draft?.type ?? 'MESSAGE'
+})
```

Comment:
```suggestion
const emptyMessage = {
  content: '',
  recipientIds: [],
  recipientNames: [],
  recipientAccountIds: [],
  title: '',
  type: 'MESSAGE' as const
}

const getInitialMessage = (
  draft: DraftContent | undefined,
  senderId: UUID
): Message =>
  draft
    ? { ...draft, senderId, recipientAccountIds: [] }
    : { senderId, ...emptyMessage }
```

Something like this would be cleaner.

Reply:
Sure would! 👍